File: tests/analyzer/invert_cond.c
#include <stdio.h>
#include <stdlib.h>

void func1() {
    int i;
    if (i < 5) {
        printf("1\n");
        exit(0);
    }
    printf("2\n");
    return;
}

void func2() {
    int i;
    if (i < 5) {
        printf("1\n");
    }
    printf("2\n");
    return;
}

int main() {
    int i, j, k, l, m;

    if (i != 0) {
        if (j != 1) {
            if (k != 2) {
                printf("1\n");
            }
            else {
                printf("2\n");
            }
        }
        else {
            printf("3\n");
        }
    }
    else {
        printf("4\n");
    }

    func1();
    func2();
    return 0;
}

File: tests/analyzer/invert_cond.py
#!/usr/bin/env python3
main = api.get_addr_from_symbol("main")
func1 = api.get_addr_from_symbol("func1")
func2 = api.get_addr_from_symbol("func2")

api.set_function(main)

api.decompile(main).print()
api.decompile(func1).print()
api.decompile(func2).print()

jumps = [0x6f2, 0x6f8, 0x6fe, 0x696, 0x6c9]

for ad in jumps:
    api.invert_cond(ad)

api.decompile(main).print()
api.decompile(func1).print()
api.decompile(func2).print()
